tests/shadow/test_shadow_measurement.py

```diff
@@ -0,0 +1,240 @@
+    """
+    Return a QNode that prepares the |00..0> state and performs a classical shadow measurement
+    """
+    if device is not None:
+        dev = qml.device(device, wires=wires, shots=shots)
+    else:
+        dev = qml.device("default.qubit", wires=wires, shots=shots)
+
+        # make the device call the superclass method to switch between the general qubit device and device specific implementations (i.e. for default qubit)
+        dev.classical_shadow = super(type(dev), dev).classical_shadow
+
+    @qml.qnode(dev, interface=interface)
+    def circuit():
+        return qml.classical_shadow(wires=range(wires))
+
+    return circuit
+
+
+class TestShadowMeasurement:
+    """Unit tests for classical_shadow measurement"""
+
+    wires_list = [1, 2, 3, 4]
+    shots_list = [1, 10, 100]
+    seed_recipes_list = [True, False]
+
+    @pytest.mark.parametrize("wires", wires_list)
+    @pytest.mark.parametrize("shots", shots_list)
+    @pytest.mark.parametrize("seed", seed_recipes_list)
+    def test_measurement_process_numeric_type(self, wires, shots, seed):
+        """Test that the numeric type of the MeasurementProcess instance is correct"""
+        res = qml.classical_shadow(wires=range(wires), seed_recipes=seed)
+        assert res.numeric_type == int
+
+    @pytest.mark.parametrize("wires", wires_list)
+    @pytest.mark.parametrize("shots", shots_list)
+    @pytest.mark.parametrize("seed", seed_recipes_list)
+    def test_measurement_process_shape(self, wires, shots, seed):
+        """Test that the shape of the MeasurementProcess instance is correct"""
+        dev = qml.device("default.qubit", wires=wires, shots=shots)
+        res = qml.classical_shadow(wires=range(wires), seed_recipes=seed)
+        assert res.shape(device=dev) == (2, shots, wires)
+
+        # test an error is raised when device is None
+        msg = "The device argument is required to obtain the shape of a classical shadow measurement process"
+        with pytest.raises(qml.measurements.MeasurementShapeError, match=msg):
+            res.shape(device=None)
+
+    @pytest.mark.parametrize("wires", wires_list)
+    @pytest.mark.parametrize("shots", shots_list)
+    @pytest.mark.parametrize("seed", seed_recipes_list)
+    def test_measurement_process_copy(self, wires, shots, seed):
+        """Test that the attributes of the MeasurementProcess instance are
+        correctly copied"""
+        dev = qml.device("default.qubit", wires=wires, shots=shots)
+        res = qml.classical_shadow(wires=range(wires), seed_recipes=seed)
+
+        copied_res = copy.copy(res)
+        assert type(copied_res) == type(res)
+        assert copied_res.return_type == res.return_type
+        assert copied_res.wires == res.wires
+        assert copied_res.seed == res.seed
+
+    @pytest.mark.all_interfaces
+    @pytest.mark.parametrize("wires", wires_list)
+    @pytest.mark.parametrize("shots", shots_list)
```

Comment:
I'd prefer to have tests for `wires=1` and `wires>1`, and likewise for `shots`. Previously we found bugs related to the output being squeezed for arrays of length 1.

Reply:
`wires=1` and `wires>1` I agree, but maybe `[1,2,3,4]` is not necessary and `[1,3]` is enough?
Can you remind me what the `shots=1` test is doing? Can we do `shots = [1,100]` for unit tests? And one fix number of shots for the integration tests?

Reply:
`shots=1` is also testing that the output shape is correct (`(2, 1, wires)` instead of `(2, wires)` from before). I can change the shots list to `[1, 100]`.

Mind you, the vast majority of the test time is spend in `test_return_distribution`, where `shots=1000` is used. I don't think there's a way around this since we need a high number of shots to accurately estimate the distribution.

Reply:
Maybe we should reconsider keeping this test in the first place. We are kind of testing whether numpy random uniform is actually uniform if I understand correctly?

I dont know what futuristic machine Albert is running his tests on but on my laptop its more like 10 minutes than 10 seconds 😄 🚀 And that I think is indeed a problem.

Reply:
This test is a sanity check for whether the results are what we expect them to be. Sure, if you read the implementation, then you might think that the test just checks if numpy random works correctly. If the implementation changes in the future, however, then it's good to have this test for sanity.

I think we're missing the bigger picture here. The later tests will almost certainly take longer, since they require a high number of shots. There isn't an easy way around this. Personally I'd longer and more thorough testing. 

(FYI, these tests run in around 6s on my laptop 😄)

Reply:
Oh wow is WSL really that terrible performance wise? Or is my environment crooked? Takes me `476 passed, 6 warnings in 47.71s`. The run earlier must have gotten caught up somewhere, I canceled it after 7 minutes at 70% but was also on one of the later branches (`shadow_entropies`).